services/debugger/source/cache.go

```diff
@@ -0,0 +1,109 @@
+package sourcedebugger
+
```

Comment:
Can we move all the cache layers onto the top(debugger level) if we are repeating the same code at multiple places.Or is there a specific reason we did this?

Reply:
done